Mathlib/CategoryTheory/MorphismProperty/Composition.lean

```diff
@@ -198,6 +198,86 @@ instance inf {P Q : MorphismProperty C} [P.IsMultiplicative] [Q.IsMultiplicative
 
 end IsMultiplicative
 
+/-- Given a morphism property `W`, the `MultiplicativeClosure W` is the smallest
+multiplicative property greater or equal than `W`. -/
+inductive MultiplicativeClosure (W : MorphismProperty C) : MorphismProperty C
+  | of {x y : C} (f : x ⟶ y) (hf : W f) : MultiplicativeClosure W f
+  | id (x : C) : MultiplicativeClosure W (𝟙 x)
+  | comp_of {x y z : C} (f : x ⟶ y) (g : y ⟶ z) (hf : MultiplicativeClosure W f) (hg : W g) :
+    MultiplicativeClosure W (f ≫ g)
+
+/-- A variant of MultiplicativeClosure in which compositions are taken on the left rather than
+on the right. We show in `MultiplicativeClosure.eq_prime` that this is in fact equal
+to MultiplicativeClosure. -/
+inductive MultiplicativeClosure' (W : MorphismProperty C) : MorphismProperty C
```

Comment:
My suggestion is not to define `MultiplicativeClosure'`, but to provide an alternative induction principle.

Reply:
It may be necessary or convenient to define `MultiplicativeClosure'` to prove this, so maybe the docstring should make clear that the better API is `multiplicativeClosure`, but in case the user needs the other induction principle (and basically only in this case), they can use `eq_prime`.

Reply:
Yeah, it seems that proving directly the induction principle without this helper isn’t that straightforward... I’ll leave it and document it as you say.